test_rclcpp/CMakeLists.txt

```diff
@@ -182,14 +182,29 @@ if(BUILD_TESTING)
     else()
       set(target_suffix "__${rmw_implementation1}__${rmw_implementation2}")
     endif()
-    # TODO(hidmic): enable tests once FastRTPS <=> (Opensplice|Connext) interoperation
-    # issues are resolved (see https://github.com/ros2/rmw_fastrtps/issues/246).
-    custom_launch_test_two_executables(test_node_name
-      node_with_name node_name_list
-      ARGS1 "${rmw_implementation1}" ARGS2 "node_with_name_${rmw_implementation1}"
-      RMW1 ${rmw_implementation1} RMW2 ${rmw_implementation2}
-      TIMEOUT 15
-      ${SKIP_TEST})
+
+    set(rmw_implementation1_is_fastrtps FALSE)
+    set(rmw_implementation2_is_fastrtps FALSE)
+    if(rmw_implementation1 MATCHES "(.*)fastrtps(.*)")
+      set(rmw_implementation1_is_fastrtps TRUE)
+    endif()
+    if(rmw_implementation2 MATCHES "(.*)fastrtps(.*)")
+      set(rmw_implementation2_is_fastrtps TRUE)
+    endif()
+
+    # Skipped tests between fastrtps and others, as the node names are communicated with a different mechanism.
+    # TODO(ivanpauno): Reenable this tests after the other implementations also use on Participant per Context.
+    if(
+      (NOT rmw_implementation1_is_fastrtps AND NOT rmw_implementation2_is_fastrtps) OR
+      (rmw_implementation1_is_fastrtps AND rmw_implementation2_is_fastrtps)
+    )
+      custom_launch_test_two_executables(test_node_name
+        node_with_name node_name_list
+        ARGS1 "${rmw_implementation1}" ARGS2 "node_with_name_${rmw_implementation1}"
+        RMW1 ${rmw_implementation1} RMW2 ${rmw_implementation2}
+        TIMEOUT 15
+        ${SKIP_TEST})
```

Comment:
Any reason to not just set SKIP_TEST in the skipped cases and call custom_launch_test... unconditionnally ?

Reply:
The `SKIP_TEST` flag does generate the tests, and they are actually run but without doing anything (see [here](https://github.com/ament/ament_cmake/blob/a4aa81a6ef2e61e5ea684c929a4c45080a9fe1db/ament_cmake_test/cmake/ament_add_test.cmake#L88)).
In this way, the tests aren't generated at all, which I prefer.

Reply:
Fair enough.

My rationale was because I like the idea that not supported communication combinations can be easily seen by just going to see the [list of skipped tests](https://ci.ros2.org/view/nightly/job/nightly_linux_release/1494/testReport/(root)/test_communication/). But it may not be relevant in this case